src/components/card/horizontal-video-card/horizontal-video-card.jsx

```diff
@@ -0,0 +1,64 @@
+import "./horizontal-video-card.css"
+import { useVideo } from "contexts/video-context";
+import { useState, useEffect, useRef } from "react"
+import { Link, useNavigate } from "react-router-dom";
+
+const HorizontalVideoCard = ({videoDetails}) => {
+    const {videoClickHandler} = useVideo();
+    const [dropdown,setDropdown] = useState(false)
+    const {_id,title,creator,thumbnail,views,uploadTime,duration,embedId} = videoDetails
+    const navigate = useNavigate();
+
+    const dropdownOptions = [
+        {optionText:"Add to watch later",optionIcon:"clock"},
+        {optionText:"Add to Playlist",optionIcon:"circle-play"},
+        {optionText:"Remove from watch later",optionIcon:"trash"}]
+    
+    const settingRef = useRef()
+    useEffect(()=>{
+        document.addEventListener("mousedown",(e)=>{
+            if(!settingRef.current?.contains(e.target)){
+            setDropdown(false)}})
+    })
+
+    const clickHandler = () => {
+        videoClickHandler(videoDetails)
+        navigate(`/watch/${_id}`)
+    }
+
+    return (<div className="m1 card horizontal-card text-decor-none">
+        <div className="flex-r">
+            <Link to={`/watch/${embedId}`} onClick={()=>clickHandler()}>
```

Comment:
If there is no parameter in the function, you can directly call it by `onClick={clickHandler}`, instead of using a callback..

Reply:
Done